src/DiseaseStatus.H

```diff
@@ -217,8 +201,8 @@ void DiseaseStatus<AC,ACT,ACTD,A>::updateAgents(AC& a_agents, /*!< Agent contain
             ParallelFor( np, [=] AMREX_GPU_DEVICE (int i) noexcept
             {
                 if (flag_hosp_ptr[i] == 1) {
-                    AMREX_ALWAYS_ASSERT(!isHospitalized(i, ptd));
-                    assign_hospital( i, hosp_i_ptr, hosp_j_ptr, ptd);
+                    // could have already been assignedto a hospital from another disease
+                    if (!isHospitalized(i, ptd)) assign_hospital( i, hosp_i_ptr, hosp_j_ptr, ptd);
```

Comment:
I don't think the agent has already been assigned to a hospital. We are now outside the `for (int d = 0; d < n_disease; d++ ) {` loop. After it iterates through all diseases, if the `flag_hosp_ptr[i]` is set to `1` (i.e., marked for hospitalization for at least one disease), the agent gets assigned a hospital.

Reply:
Even without that fix, I don't see how an agent that `AMREX_ALWAYS_ASSERT` would fail, especially since you are probably seeing that fail for a single disease, right? I am staring at that piece of code in both the `development` branch and your `UrbanPop` branch. Given that the incubation period, once sampled from a normal distribution, is fixed for an agent, I don't see why it will enter the `counter_ptr[i] == Math::floor(incubation_period_ptr[i])` more than once.

Reply:
This could be a deeper bug. Once my fix is merged, can you put back that assert and see if it fails? If it does, can you share that input file so that I can look into it?

Reply:
Yes, likely a deeper bug that I never figured out. I'll merge your fix into UrbanPop and check this again.

Reply:
Turns out i was actually testing with 2 diseases when this assertion got triggered. It doesn't for one disease.

Reply:
Okay, then I definitely need to look into this. Let me know once you are done with all the changes you want to make in this branch, and I'll try it out.

Reply:
I figured out the reason for this! The logic that I implemented for checking if an agent needs to be hospitalized is not correct for multiple diseases. I inadvertently assumed that all the diseases will have the same incubation period for an agent. So, what you have done here is fine for now, but I need to improve the hospital model.